DBSCE/LocalKeyHelper-Mac.md

```diff
@@ -0,0 +1,96 @@
+<span style="color:red">!!! DRAFT - Please note this page is still in draft as we work on platform specific details
+</span>
+
+### Local key helper on Mac
+
+On macOS, a Local Key Helper is a protocol (`LocalKeyHelper.h`), and its implementation (`LocalKeyHelperImpl.h`/`LocalKeyHelperImpl.m`) is yet to be determined.
+
+The Local Key Helper will be implemented and shipped as an XPC service on macOS. It will be signed with app sandbox entitlement for securety requirement.
+
+To start the Local Key Helper XPC service, the service plist (property list) needs to be defined and registered in the launchAgent. This will ensure the service starts up when a idP sends a service connections.
+
+Here is an example of the Local Key Helper XPC service plist:
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+    <key>Label</key>
+    <string>com.contoso.LocalKeyHelper</string>
+    <key>Program</key>
+    <string>/pathToXcpService/LocalKepHelper.xpc/Contents/MacOS/LocalKepHelper</string>
+    <key>KeepAlive</key>
+    <true/>
+    <key>MachServices</key>
+    <dict>
+        <key>com.contoso.LocalKeyHelper</key>
+        <true/>
+    </dict>
+</dict>
+</plist>
+```
+
+When the browser (e.g. Chrome) communicates with this service, it needs to establish a service connection first.
+Here is an example of starting a new connection with the Local Key Helper:
+
+```objective-c
+// Start a new connection by providing the Local Key Helper defined in the plist
+NSXPCConnection* localKeyhelper = [[NSXPCConnection alloc] initWithMachServiceName:@"com.contoso.LocalKeyHelper"];
+
+// Make the connection conform to the defined Local Key Helper protocol
+[localKeyhelper setRemoteObjectInterface:[NSXPCInterface interfaceWithProtocol:@protocol(LocalKeyHelper)]];
+```
+
+The browser (e.g. Chrome) can then call the methods defined in the LocalKeyHelper protocol, and the implementations will be provided by the XPC service.
+
+**Note:**
+
+- It is important to define the LocalKeyHelper protocol exactly the same way in both the browser and the XPC service provider. This allows for easy extension of the protocol in the future (e.g. using a dictionary as an input parameter instead of a strict object type).
+- The input and output parameters of the XPC service should follow the `NSSecureCoding` protocol.
+- Un-sandboxed applications can communicate directly to the XPC service.
+
+#### Define Available Local Key Helpers
```

Comment:
Let's rename to "Browser discovery process"?

Also, I think we should provide similar structure for all 3 platforms. Could you help aligning the headings between 3 platform docs?

Reply:
Okay, let us track this as separate PRs. I will remove them for now and we can review the specifics separately.

Reply:
Done